subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ApplicationPluginIntegrationTest.groovy

```diff
@@ -497,4 +498,20 @@ rootProject.name = 'sample'
         File scriptOutputDir = file('build/scripts')
         new File(scriptOutputDir, filename)
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/1923")
+    def "not up-to-date if classpath changes"() {
```

Comment:
need some test coverage that when the _content_ of the classpath changes, but the classpath does not, then the task is up-to-date